Lib/test/test_typing.py

```diff
@@ -4300,22 +4300,23 @@ def test_basics_functional_syntax(self):
         self.assertEqual(Emp.__total__, True)
 
     def test_basics_keywords_syntax(self):
-        Emp = TypedDict('Emp', name=str, id=int)
-        self.assertIsSubclass(Emp, dict)
-        self.assertIsSubclass(Emp, typing.MutableMapping)
-        self.assertNotIsSubclass(Emp, collections.abc.Sequence)
-        jim = Emp(name='Jim', id=1)
-        self.assertIs(type(jim), dict)
-        self.assertEqual(jim['name'], 'Jim')
-        self.assertEqual(jim['id'], 1)
-        self.assertEqual(Emp.__name__, 'Emp')
-        self.assertEqual(Emp.__module__, __name__)
-        self.assertEqual(Emp.__bases__, (dict,))
-        self.assertEqual(Emp.__annotations__, {'name': str, 'id': int})
-        self.assertEqual(Emp.__total__, True)
+        with self.assertWarns(DeprecationWarning):
+            Emp = TypedDict('Emp', name=str, id=int)
+            self.assertIsSubclass(Emp, dict)
+            self.assertIsSubclass(Emp, typing.MutableMapping)
+            self.assertNotIsSubclass(Emp, collections.abc.Sequence)
+            jim = Emp(name='Jim', id=1)
+            self.assertIs(type(jim), dict)
+            self.assertEqual(jim['name'], 'Jim')
+            self.assertEqual(jim['id'], 1)
+            self.assertEqual(Emp.__name__, 'Emp')
+            self.assertEqual(Emp.__module__, __name__)
+            self.assertEqual(Emp.__bases__, (dict,))
+            self.assertEqual(Emp.__annotations__, {'name': str, 'id': int})
+            self.assertEqual(Emp.__total__, True)
```

Comment:
This is a very large block of code to have under a single `self.assertWarns(DeprecationWarning)`. Surely the only part of this that we want to raise a `DeprecationWarning` is the actual definition of the `TypedDict`? The rest can be as it was before, no?